gist.go

```diff
@@ -0,0 +1,92 @@
+package model
+
+// Typed constants. Types are not exported as you can still export the values 
+// for use in `if` and `switch` statements.
+type channelType string
+
+// CamelCase Constant Names
+// Constants are exported for use in `if` and `switch` statements and similar in other packages.
+const (
+	ChannelTypeOpen    channelType = "O"
+	ChannelTypePrivate channelType = "P"
+)
+
+type Channel struct {
+	// Member variables are not exported to ensure immutability.
+	id       string `json:"id"`
+	name     string `json:"name"`
+	createAt int64  `json:"create_at"`
+	// etc...
+
+	// Question: what do we do when there's a semantically meaningful
+	// "null" value for a member variable?
+	schemeID *string `json:"scheme_id"`
+
+	// Question: is this the best way of storing props and similar map/list data?
+	props map[string]interface{} `json:"props" db:"-"`
+}
+
+// Getters to access the member variables are provided. These *never* return pointers.
+// Correct capitalisation of initialisms
+func (channel *Channel) ID() string {
+	return channel.id
+}
+
+func (channel *Channel) Name() string {
+	return channel.name
+}
+
+// Returning `interface{}` feels gross. Can we do better than this?
```

Comment:
See https://github.com/mattermost/mattermost-server/pull/14301/files#r415115283. We won't have prop getters, instead have proper field accessors. Also, having "Get' prefix for a getter is un-idiomatic. Just the field name is good enough.

Reply:
And assuming we still need the props field to give flexibility to the client, any thoughts on how to solve this?

Reply:
A good improvement would be using `ok` return. Like `(interface{}, bool)`

Reply:
> A good improvement would be using `ok` return. Like `(interface{}, bool)`

Just bear in mind that if we do that we won't be able to type assert directly like we do almost everywhere right now with `model.Post` props.

Reply:
Yeah, I think we faced a similar situation in load tests too. It's gross, and that's why we should try to avoid it as much as possible :stuck_out_tongue:

Reply:
So any new props that get added must have a field accessor then, yes?

Reply:
Does not look like that as per the conversation above. We will still keep using props for plugins and such. So for that, we will still keep the prop accessor. But all internal state will move to the struct.